src/GitHub.App/Models/RepositoryHosts.cs

```diff
@@ -157,11 +157,15 @@ public IRepositoryHost LookupHost([AllowNull] HostAddress address)
                     );
                     if (successful)
                     {
+                        // We need to add the connection before assigning to GitHubHost or
+                        // EnterpriseHost so that anything listening to those will find the
+                        // associated connection.
+                        connectionManager.AddConnection(address, usernameOrEmail);
```

Comment:
We'll probably have to set the host fields (without raising notifications), then call `AddConnection`, and then raise the field set events. We don't _really_ need `RaiseAndSetIfChanged` on the `GitHubHost/EnterpriseHost`, it's not something that gets set a lot, so we can probably just do `field = value; this.RaisePropertyChange()` on those setters.